### ISSUE TITLE
Build-time error on OSX, when building with 32 bit Erlang, is confusing.

### ISSUE THREAD

**Issue by elight** [basho/riak#364]
When building Riak with a 32 bit Erlang on OSX results in an error like:

```
> in16.h ../../dist/include/nspr
> cd src; make export
> cd io; make export
> make[6]: Nothing to be done for `export'.
> cd linking; make export
> cc -o prlink.o -c     -m32  -Wall -fno-common -pthread -O2 -fPIC  -UDEBUG  -DNDEBUG=1 -DXP_UNIX=1 -DDARWIN=1 -DHAVE_BSD_FLOCK=1 -DHAVE_SOCKLEN_T=1 -DXP_MACOSX=1 -DHAVE_LCHOWN=1 -DHAVE_STRERROR=1  -DFORCE_PR_LOG -D_PR_PTHREADS -UHAVE_CVAR_BUILT_ON_SEM -D_NSPR_BUILD_ -I../../../dist/include/nspr -I../../../pr/include -I../../../pr/include/private -I/Developer/Headers/FlatCarbon  prlink.c
> prlink.c:48:10: fatal error: 'CodeFragments.h' file not found
> #include <CodeFragments.h>
>          ^
> 1 error generated.
> make[6]: *** [prlink.o] Error 1
> make[5]: *** [export] Error 2
> make[4]: *** [export] Error 2
> make[3]: *** [export] Error 2
> make[2]: *** [/Users/jeff/Database/riaka-1.2.0/deps/erlang_js/c_src/system/lib/libnspr4.a] Error 2
> make[1]: *** [c_src] Error 2
> ERROR: Command [compile] failed!
> make: *** [rel] Error 1
```

[This](http://lists.basho.com/pipermail/riak-users_lists.basho.com/2012-August/009110.html) email to the mailing list explains the issue.  It would be helpful if the build process could provide a suggestion to the user that, on encountering this error, that they are likely running a 32 bit Erlang. It would likely save several users some agony.

**Comment by jaredmorrow** [comment]
This is not related to 32bit Erlang.  We do not support 32bit, but you can still build Riak on 32bit just fine.  This is a general problem people ran into finding libraries installed by XCode.

**Comment by elight** [comment]
While I realize what the error indicated does not appear to correlate with
32bit Erlang, per @cmeik's advice, I rebuilt my Erlang to 64bit and the
compilation error vanished.

Why this should be makes no sense to me either…

The missing header files were present on my machine; however, they were
under significantly different paths than any supplied in the include paths
to GCC.

On Sun, Aug 18, 2013 at 9:34 AM, Jared Morrow notifications@github.comwrote:

> This is not related to 32bit Erlang. We do not support 32bit, but you can
> still build Riak on 32bit just fine. This is a general problem people ran
> into finding libraries installed by XCode.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/basho/riak/issues/364#issuecomment-22830545
> .

## 

Evan Light
http://evan.tiggerpalace.com
@elight

**Comment by jaredmorrow** [comment]
So I don't think this is a issue with Riak, but an issue with how OSX treats library paths.  I'm all for making errors more obvious, but in this case I don't know what we'd change in Riak.

**Comment by cmeiklejohn** [comment]
It's something specifically related to nspr (Netscape Portable Runtime) in erlang_js; some sort of environment change is happening when building with 64-bit clang, causing these files to no longer be available in the search path.

**Comment by cmeiklejohn** [comment]
Confirmed this error does not occur when building erlang_js outside of Riak.

**Comment by cmeiklejohn** [comment]
Resolved in basho/erlang_js#39.

**Comment by cmeiklejohn** [comment]
Tracking in basho/erlang_js#39.
